lib/omniauth/strategies/shopify.rb

```diff
@@ -23,6 +23,31 @@ def valid_site?
         !!(/\A(https|http)\:\/\/[a-zA-Z0-9][a-zA-Z0-9\-]*\.#{Regexp.quote(options[:myshopify_domain])}[\/]?\z/ =~ options[:client_options][:site])
       end
 
+      def valid_signature?
+        return false unless request.POST.empty?
+
+        params = request.GET
+        signature = params['hmac']
+        timestamp = params['timestamp']
+        return false unless signature && timestamp
+
+        return false unless timestamp.to_i > Time.now.to_i - 5 * 60
```

Comment:
I was thinking of the case where the server running the app has a slow clock (rather than one of Shopify's servers). But if replay attacks aren't a concern for this request then it's a moot point.